include/platform-aws.h

```diff
@@ -58,4 +59,13 @@ struct ec2_platform_data *platform_aws_get_platform_entry(const char *platform_t
 							  struct ec2_platform_data *platform_data_list,
 							  size_t platform_data_len);
 
+class Aws : public Platform {
```

Comment:
The things above should also be member functions.  They can even be protected and you can play the same inheritance for unit test trick we play elsewhere.